src/Libraries/RevitNodes/Elements/Room.cs

```diff
@@ -0,0 +1,220 @@
+        /// Internal reference to the Revit Element
+        /// </summary>
+        internal Autodesk.Revit.DB.Architecture.Room InternalRevitElement
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalRevitElement; }
+        }
+       
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Create from existing element
+        /// </summary>
+        /// <param name="Room"></param>
+        internal Room(Autodesk.Revit.DB.Architecture.Room Room)
+        {
+            SafeInit(() => InitElement(Room));
+        }
+
+        /// <summary>
+        /// Create a new Room by Level and location
+        /// </summary>
+        /// <param name="level"></param>
+        /// <param name="location"></param>
+        /// <param name="name"></param>
+        /// <param name="number"></param>
+        private Room(Autodesk.Revit.DB.Level level, Autodesk.Revit.DB.XYZ location, string name, string number)
+        {
+            SafeInit(() => Init(level, location, name, number));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        /// Set internal element
+        /// </summary>
+        private void InitElement(Autodesk.Revit.DB.Architecture.Room element)
+        {
+            InternalSetElement(element);
+        }
+
+        private void Init(Autodesk.Revit.DB.Level level, Autodesk.Revit.DB.XYZ location, string name, string number)
+        {
+            // Get document and open transaction
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            // Get existing room element if possible
+            var RoomElem = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.Architecture.Room>(document);
+
+            if (RoomElem == null)
+            {
+                // Create new Room element
+                RoomElem = document.Create.NewRoom(level, new Autodesk.Revit.DB.UV(location.X, location.Y));
+            }
+            else
+            {
+                // Update Location only
+                Autodesk.Revit.DB.LocationPoint point = (Autodesk.Revit.DB.LocationPoint)RoomElem.Location;
+                point.Point = location;
+            }
+
+            // Apply name and number if set
+            if (name != "") { RoomElem.Name = name; }
+            if (number != "") { RoomElem.Number = number; }
+
+            InternalSetElement(RoomElem);
+
+            // Commit transaction
+            TransactionManager.Instance.TransactionTaskDone();
+
+            if (RoomElem != null)
+            {
+                ElementBinder.CleanupAndSetElementForTrace(document, this.InternalElement);
```

Comment:
so I'm a bit confused by this logic - can you clarify why if the room element is not null, meaning it was been created or pulled from trace that you then destroy it? and the else statement seems it will never be true...

Can a room not be rebound? I would think that you should get rid of the else statement and use SetElement for trace here instead.

`CleanupAndSetElement` destroys the element - meaning that the ID of the room will change if it was pulled from trace right?